applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/high_cycle_fatigue_law_integrator.h

```diff
@@ -0,0 +1,322 @@
+    {
+        array_1d<double,3> principal_stresses;
+        ConstitutiveLawUtilities<6>::CalculatePrincipalStresses(principal_stresses, rStressVector);
+
+
+        double abs_component = 0.0, average_component = 0.0, sum_abs = 0.0, sum_average = 0.0;
+        for (unsigned int i = 0; i < principal_stresses.size(); ++i) {
+            abs_component = std::abs(principal_stresses[i]);
+            average_component = 0.5 * (principal_stresses[i] + abs_component);
+            sum_average += average_component;
+            sum_abs += abs_component;
+        }
+        const double pre_indicator = sum_average / sum_abs;
+        if (pre_indicator < 0.5) {
+            rFactor = -1.0;
+        } else {
+            rFactor = 1.0;
+        }
+    }
+
+    /**
+     * @brief This method computes de reversion factor 
+     * @param MaxStress Signed maximum equivalent stress in the current cycle.
+     * @param MinStress Signed minimum equivalent stress in the current cycle. 
+     */
+    static void CalculateReversionFactor(const double MaxStress, const double MinStress, double& rReversionFactor)
+    {
+        rReversionFactor = MinStress / MaxStress;
+    }
+
+    /**
+     * @brief This method computes the reduction factor used in the high cycle fatigue model 
+     * @param MaxStress Signed maximum stress in the current cycle.
+     * @param MinStress Signed minimum stress in the current cycle.
+     * @param ReversionFactor Ratio between the minimum and maximum signed equivalent stresses for the current load cycle.
+     * @param MaterialParameters Material properties.
+     * @param NumbreOfCycles Number of cycles at the current step.
+     * @param FatigueReductionFactor Reduction factor from the previous step to be reevaluated.
+     * @param B0 Internal variable of the fatigue model used for the calculation of the FatigueReductionFactor.
+     */
+    static void CalculateFatigueReductionFactor(const double MaxStress,
+                                                const double MinStress,
+                                                double& rReversionFactor,
+                                                const Properties& rMaterialParameters,
+                                                const double NumbreOfCycles,
+                                                double& rFatigueReductionFactor,
+                                                double& rB0
+                                                )
+	{
+        CalculateReversionFactor(MaxStress, MinStress, rReversionFactor);
+        const Vector& r_fatigue_parameters = rMaterialParameters[HIGH_CYCLE_FATIGUE_PARAMETERS];
+        const double yield_stress = rMaterialParameters.Has(YIELD_STRESS) ? rMaterialParameters[YIELD_STRESS] : rMaterialParameters[YIELD_STRESS_TENSION];
+
+        const double Se = r_fatigue_parameters[0] * yield_stress;
+        const double STHR1 = r_fatigue_parameters[1];
```

Comment:
Please use a more descriptive name for the variables

Reply:
I will try it, but these are semi-empirical variables used for buillding the S-N curves and do not have a clear definition. I have used the nomenclature used in the paper A continuum mechanics model for mechanical fatigue analysis, (S. Oller, 2005),

Reply:
In the paper is described as STHR1?